core/service/src/lib.rs

```diff
@@ -869,33 +827,12 @@ impl<C: Components> network::TransactionPool<ComponentExHash<C>, ComponentBlock<
 /// A provider of current authority key.
 #[derive(Clone)]
 pub struct AuthorityKeyProvider {
-	roles: Roles,
-	keystore: Option<Arc<Keystore>>,
-	password: crypto::Protected<String>,
+	keystore: Arc<Mutex<Store>>,
 }
 
 impl offchain::AuthorityKeyProvider for AuthorityKeyProvider {
 	fn authority_key<TPair: Pair>(&self) -> Option<TPair> {
-		if self.roles != Roles::AUTHORITY {
-			return None
-		}
-
-		let keystore = match self.keystore {
-			Some(ref keystore) => keystore,
-			None => return None
-		};
-
-		let loaded_key = keystore
-				.contents()
-				.map(|keys| keys.get(0)
-					.map(|k| keystore.load(k, self.password.as_ref()))
-				);
-
-		if let Ok(Some(Ok(key))) = loaded_key {
-			Some(key)
-		} else {
-			None
-		}
+		None
```

Comment:
@tomusdrw: I'm not sure how to implement this since it doesn't mingle well with the design of this PR.

- session key transitions are the responsability of aura/babe/grandpa
- the keystore broadcast new keys to local keystores
- after a session key transition aura/babe/grandpa remove the old session key from their local keystore

Reply:
Offchain workers should basically get the same session key that is used for everything else. I suppose it will require querying the runtime (latest block) to do an intersection between local keys and the ones set there.